Madness/String.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  String.swift
+//  Madness
+//
+//  Created by Josh Vera on 10/19/15.
+//  Copyright © 2015 Rob Rix. All rights reserved.
+//
+
+import Foundation
+import Prelude
+
+public typealias CharacterParser = Parser<String.CharacterView, Character>.Function
+public typealias CharacterArrayParser = Parser<String.CharacterView, [Character]>.Function
+public typealias StringParser = Parser<String.CharacterView, String>.Function
+public typealias DoubleParser = Parser<String.CharacterView, Double>.Function
+
+public typealias IntParser = Parser<String.CharacterView, Int>.Function
+
+func maybePrepend<T>(value: T?) -> [T] -> [T] {
+	return { value != nil ? [value!] + $0 : $0 }
```

Comment:
This might be cleaner using `Optional.map`, `??`, and curried function syntax:

```swift
func maybePrepend<T>(value: T?)(_ rest: [T]) -> [T] {
	return (value.map { [ $0 ] } ?? []) + rest
}
```